### ISSUE TITLE
[homematic] ThingState "Config Pending" if FirmwareUpdate is available, not "Config Pending"

### ISSUE THREAD

**Issue by KaaNee** [openhab/openhab-addons#15430]
## Expected Behavior
The state "config pending" is an defined state of homematic, when a device is waiting for a config-update from the homematic-hub (and it's not transfered yet).

## Current Behavior
Additionally the state "config pending" is displayed, if a firmware update for a device is available.
This is NOT "config pending" - in my opinion. The homematic-hub does not have the state "config pending" for these items (service-message). 
A possible (!) firmware update is displayed at another position in homematic itself. (or only raspimatic ?)

Attention: A started firmware update in homematic-firmware-management (pressing "update" for a device) results in a service-message "FIrmware update pending" - like the service-message when "really" a config is pending.

## Possible Solution
Either display another state in openhab "firmware update available"
OR
ignore new outstanding firmware updates and display "online" - because the thing itself is normal usable.
As i remember correctly other bindings (like mqtt-items) have an extra channel "firmware update available", isn't it ?

## Steps to Reproduce (for Bugs)
1. Have a device in homematic which has a firmware update available
2. See things-overview in openhab - "config pending" as state is displayed.

I saw this behaviour after a restart of openhab, when the thing/itemstate is fetched completely new.

## Context
This is more a information/cosmetic bug. it doesn't affect the usage, but irritates the user.

## Your Environment
openhab 3.4.1
homematic / raspberrimatic Version 3.67.10.20230225

**Comment by lsiepel** [comment]
If you have some traffic details (binding trace log level) it mightbe helpfull to detect this state and map the thing status accordingly.

**Comment by KaaNee** [comment]
Hi @lsiepel . Short answer: No.
I don't have openhab in use anymore.

I'm sorry, can't help here.

**Comment by lsiepel** [comment]
Ik, thanks for taking the time to answer.
@maniac103 maybe we can consider to look at this after the refactoring of the binding

**Comment by maniac103** [comment]
I'm not even sure whether we should fix this/change behavior here. Yes, the binding conflates two different states (config pending / update available) into one, but ThingStatusDetail [doesn't have](https://www.openhab.org/javadoc/latest/org/openhab/core/thing/thingstatusdetail) for 'update available', so the current solution seems to make sense to me. I'm not sure removing that information actually improves UX. Maybe we should augment the state with a description text?

**Comment by KaaNee** [comment]
Hey @maniac103 , i'm with you, diff the state using different text seems enough. My opinion.

Greetings Kai.

**Comment by lsiepel** [comment]
I know @andrewfg is working on a pr related to firmware updates. Wdyt about the thing state?

**Comment by andrewfg** [comment]
I agree that "software update available" is not really "config pending" but is rather just an info about something potentially new. So you could consider adding a property "softwareState" = "Update Available" and/or you could take advantage of the "blue dot" marker that we recently added to MainUI .. https://github.com/openhab/openhab-webui/pull/4069 .. in which if thing status == online and thing status detail description == "something new" then a blue dot is applied to the thing status badge to indicate that the thing has something new.

---

PS #20498 is an implementation example for the Hue binding which could possibly be emulated for this binding too..

**Comment by andrewfg** [comment]
~PS I could make a PR to do that if you wish.~ => See #20658

**Comment by lsiepel** [comment]
Thanks Käi for the report, sorry it took so long. :-)
